client/me/form-base/with-form-base.jsx

```diff
@@ -0,0 +1,126 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import debugFactory from 'debug';
+
+const debug = debugFactory( 'calypso:me:form-base' );
+
+/**
+ * Internal dependencies
+ */
+import notices from 'calypso/notices';
+import user from 'calypso/lib/user';
+
+const withFormBase = ( WrappedComponent ) => {
+	return class extends React.Component {
+		static displayName = `withFormBase(${ WrappedComponent.displayName || WrappedComponent.name })`;
+
+		state = {
+			redirect: false,
+			submittingForm: false,
+			changingUsername: false,
+			usernameAction: 'new',
+			showNotice: false,
+		};
+
+		componentDidMount() {
+			this.props.userSettings.getSettings();
+		}
+
+		componentWillUnmount() {
+			// Silently clean up unsavedSettings before unmounting
+			this.props.userSettings.unsavedSettings = {};
+		}
+
+		UNSAFE_componentWillReceiveProps( nextProp ) {
+			if ( nextProp.showNoticeInitially ) {
```

Comment:
I'll check this out and see if we can get rid of this one 👍🏻

Reply:
I'm assuming we decided to keep this for the time being?

Reply:
Oops, I meant to remove this. The component which uses this will be migrated later, just need to make a note about it. Will add a commit before merging.

Reply:
Thanks!